### PR TITLE
feat: add custom api data to plan

### PR DESCRIPTION
**Pull request by Sxyntheon**
The client can now make reuqeust to TheBackend server infrastucture to retrieve personalized data from Untis. This data is also fed into the plan to get displayed

### CODE REVIEW

File: frontend/src/components/plan-components/Stundenplan.tsx [hunk 1]
```diff
@@ -26,32 +81,84 @@ export default function Stundenplan(): JSX.Element {
...
+                            )
+                        }
+                        else {
+                            if(lessons[k].substitution?.room && lessons[k].substitution?.room != "---") {
+                                roomStyle = { textDecoration: "line-through" }
+                                substitutionRoomStyle = { display: "block" }
+                            }
+                            if(lessons[k].substitution?.teacher) {
+                                teacherStyle = { textDecoration: "line-through" }
+                                if(lessons[k].substitution?.teacher != "---") {
```
Comment (ixhbinphoenix) [inline frontend/src/components/plan-components/Stundenplan.tsx:132]: This is always the case, since the API now returns null instead of ---

**Review by ixhbinphoenix — CHANGES_REQUESTED** [review]

File: frontend/src/api/main.ts [hunk 2]
```diff
@@ -1,12 +1,109 @@
...
+    try {
+        if(resultClean.body.lessons) {
+            return {
+                lessons: resultClean.body.lessons,
+                status: "200 OK",
+                message: undefined
+            }
+        }
+        else {
+            return {
```
[the remark this answers is not in the document]
Comment (Sxyntheon) [reply frontend/src/api/main.ts]: better?

File: frontend/src/logs/testStudent.ts [hunk 3]
```diff
@@ -3,232 +3,260 @@ import type { TheScheduleObject } from "../api/main"
 export const testStudent: Array<TheScheduleObject>  = [
```
[the remark this answers is not in the document]
Comment (Sxyntheon) [reply frontend/src/logs/testStudent.ts]: no

**Review by ixhbinphoenix — APPROVED** [review]
lgtm

**Review by renkertm — APPROVED** [review]
ok

**Comment by ixhbinphoenix** [comment]
damn dude, fastest merge in the west

**Pull request merged**